pkg/controller/expcertificates/issuing/issuing_controller.go

```diff
@@ -248,56 +314,14 @@ func (c *controller) failIssueCertificate(ctx context.Context, log logr.Logger,
 // issueCertificate will ensure the public key of the CSR matches the signed
 // certificate, and then store the certificate, CA and private key into the
 // Secret in the appropriate format type.
-func (c *controller) issueCertificate(ctx context.Context, log logr.Logger, nextRevision int, crt *cmapi.Certificate, req *cmapi.CertificateRequest) error {
-	csr, err := utilpki.DecodeX509CertificateRequestBytes(req.Spec.CSRPEM)
-	if err != nil {
-		return err
-	}
-
-	//Encode and issue the key-pair (store it in the Secret)
-	nextPrivateKeySecret, err := c.secretLister.Secrets(crt.Namespace).Get(*crt.Status.NextPrivateKeySecretName)
-	if apierrors.IsNotFound(err) {
-		// If secret does not exist, do nothing (keymanager will handle this).
-		return nil
-	}
-	if err != nil {
-		return err
+func (c *controller) issueCertificate(ctx context.Context, nextRevision int, crt *cmapi.Certificate, req *cmapi.CertificateRequest, pkData []byte) error {
+	secretData := secretsmanager.SecretData{
+		PrivateKey:  pkData,
```

Comment:
Tracing this back, I don't see anywhere where we are re-encoding the private key data according to `spec.keyFormat` (i.e. PKCS1 or PKCS8) - `pkData` originates from ParseTLSKeyFromSecret, which will return the raw bytes (i.e. PKCS8). If you want to leave it for now and we follow up to fix it (with some improvements to the SecretManager too) then GFI - can you add a comment and perhaps also mention it in the release note for this PR or something however? Perhaps we need a list of 'known issues' :)